src/reducers/index.ts

```diff
@@ -8,26 +8,32 @@ export default function counterReducer(
   action: ICounterAction,
 ) {
   switch (action.type) {
-    case Actions.IncreaseCount:
+    case Actions.AddCounter:
+      state.unshift({ count: 0 });
```

Comment:
For both Actions.AddCounter and Actions.RemoveCounter the state is being modified in place and then `.slice(0)` is used to generate a new object. Please do not modify any state inside Redux Reducers this can easily lead to bugs in the future. Actions.RemoveCounter can be done through `filter` and the Actions.AddCounter can be done through the es6 array spread operator.